### ISSUE TITLE
plasmoid uses icons which are dark in plasma dark theme

### ISSUE THREAD

**Issue by lbckmnn** [Martchus/syncthingtray#126]
**Relevant components**
* [ ] Standalone tray application (based on Qt Widgets)
* [x] Plasmoid/applet for Plasma desktop
* [ ] Dolphin integration
* [ ] Command line tool (`syncthingctl`)
* [ ] Integrated Syncthing instance (`libsyncthing`)
* [ ] Backend libraries

**Environment and versions**
* syncthingtray: 1.1.14
* qtutilities: 6.5.3-1
* c++utilities: propably 5.12.0-1
* Qt version: 5.15.2
* C++ compiler (name and version): gcc 11.1.0
* Operating system (name and version): arch linux with Plasma --> syncthingtray was installed from aur

**Bug description**
The icons of the Plasmoid are barely visible on breeze-dark plasma theme. 
The screenshot in the repository description shows a plasmoid with light icons. 
I guess the plasmoid just uses the wrong ones?

I also made sure that the Breeze Dark icon set is selected in system settings.
Is there an option to choose the icon set?

**Steps to reproduce**
1. set plasma theme to breeze-dark
2. install syncthingtray and place plasmoid

**Expected behavior**
plasmoid uses the appropriate icons for plasma theme.

**Screenshots**
![Screenshot_20220201_203816](https://user-images.githubusercontent.com/66389377/152044808-0463ca62-326f-4621-9763-6b7cc8299c58.png)

**Comment by Martchus** [comment]
Light colors for custom color roles (roles not provided by QPalette and hence not deducible from KDE's color settings) can be enabled in the settings.

**Comment by lbckmnn** [comment]
where exactly do i find this settings?
I would be also willing to help but i have no experience with the syncthingtray code base.

**Comment by Martchus** [comment]
Plasmoid/Tray -> Apearance -> Colors -> Bright custom text colors (use for dark color scheme)

By the way, the icon theme has no effect here because these are technically text colors.

**Comment by lbckmnn** [comment]
Yes the "Bright custom text colors" has no effekt.

Ah yes, is use the "Breeze Twighlight" global theme. 
This seems to use the breeze light theme for "Colors" but "Breeze Dark" for Plasma Style

**Comment by Martchus** [comment]
Well, not sure whether that's the case in your setup but you definitely uncovered a bug that should be fixed. Now the question is how to read colors from the Plasma theme (rather than QPalette).

**Comment by lbckmnn** [comment]
Choosing "Breeze Dark" does indeed result in white icons.
Other plasmoids (e.g. the blueetooth plasmoid) are using white icons when Plasma uses Breeze Dark but colors are set to Breeze (light). Maybe we could look there?

**Comment by Martchus** [comment]
Not really. I'm not actually rendering icons here, I'm rendering text (as the ForkAwesome icons are actually just font glyphs). All this is done in C++ and I cannot easily use the usual Qt Quick components for displaying them.

**Comment by lbckmnn** [comment]
Maybe a checkbox for dark/light color similar to the "Bright custom text colors" could be used as workaround?

**Comment by Martchus** [comment]
Looks like the answer is: https://api.kde.org/frameworks/plasma-framework/html/classPlasma_1_1QuickTheme.html

Now I only need to find a way to access it from C++.

**Comment by lbckmnn** [comment]
It is possible to just instantiate a Plasma::Theme with the default constructor. That instance will then be the currently used plasma theme. The text color can then be accessed easily. No need to use Plasma::QuickTheme.

See the minimal example:
```C++

#include "mainwindow.h"
#include <QApplication>
#include <iostream>
#include <Plasma/Theme>

int main(int argc, char *argv[])
{
    QApplication a(argc, argv);
    MainWindow w;
    Plasma::Theme currentPlasmaTheme(&w);
    std::cout << "current theme name: " << currentPlasmaTheme.themeName().toStdString() << std::endl;
    std::cout << currentPlasmaTheme.color(Plasma::Theme::TextColor).name().toStdString() << std::endl;
    w.show();
    return a.exec();
}

```

When using Plasma theme Breeze Dark the output is:

```
current theme name: breeze-dark
#fcfcfc
```
When using Breeze, the output is:
```
current theme name: default
#232627
```

I am willing to implement this in syncthingtray but i would need a couple of days because it has been a while since I last used Qt.

**Comment by Martchus** [comment]
I've just gave it a quick try as well and this works, indeed. Somehow my `QtForkAwesome::QuickImageProvider` still doesn't respect the color but I suppose I'll be able to sort it out.

**Comment by Martchus** [comment]
It should be fixed on master. (The problem why `QtForkAwesome::QuickImageProvider` didn't work was that I had the Git version of Syncthing Tray installed on my system and plasmashell's QML engine was using the image provider from there. Took me a while to figure that out, the coding was the easy part.)

**Comment by Martchus** [comment]
While https://github.com/Martchus/syncthingtray/commit/3e38a9917e497a77fa270876535fca84d940a4ff fixes the issue there's still one problem left: Theme changes aren't immediately applied (only after restarting `plasmashell`). I tried fixing this as well (see https://github.com/Martchus/syncthingtray/commit/9b7d6263e8e04a523229f302e8ee66e2f696f70b) but haven't had any luck so far. I'm not sure how to tell the Qt Quick `Image` to reload the image. Changing the source back and forth doesn't have an effect, even when caching is disabled.

**Comment by Martchus** [comment]
I'd actually still like to fix the problem mentioned in my previous comment. I just don't know how it is done in Qt Quick.

**Comment by Martchus** [comment]
The problem mentioned in https://github.com/Martchus/syncthingtray/issues/126#issuecomment-1032523273= should be fixed by 8ec133e62d757ae81dac53212053e202fb4f0a89.
